mods/default/craftitems.lua

```diff
@@ -133,11 +133,14 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 			end
 		end
 
-		local data_str = minetest.serialize(data)
-		stack:set_metadata(data_str)
+		stack:get_meta():from_table(data)
 		book_on_use(stack, player)
 	end
 
+	-- Backwards compatibility
+	stack:set_metadata("")
```

Comment:
`from_table` clears all meta before adding the new one in, thus no need to clear this field manually.